### PR TITLE
VPN-5479: metric fixes

### PR DESCRIPTION
**Pull request by mcleinman**
## Description

6 changes for issues found by QA with Super Dooper, described in the connected ticket:
1. user_changed_endpoint_geo: It missed a bunch of situations, like selecting from recents or just switching between multihop and single hop. I moved where we're tracking this, which also required a slight change in how the tutorial sets it (as we don't want to record the tutorial's changing the geo).
2. tips_and_tricks_modal_shown - had a bad merge, and some broken code from old Glean was stopping them
3. tips_and_tricks_view_opened - had a bad merge, and some broken code from old Glean was stopping them
4. using_system_language - wasn't attached to a ping, and there is seemingly no metrics ping
5. connect_on_startup_active - wasn't attached to a ping, and there is seemingly no metrics ping
6. bottom_navigation_bar_click - not properly showing the enum. Not quite sure what happened here, it had been working months ago when I put up the original code. Also deleted some unused code around here.

I also added a developer menu option to reset the "show tips and tricks modal", as there was previously no other way to show it other than uninstalling/resetting the app.

## Reference

https://mozilla-hub.atlassian.net/browse/VPN-5479

## Checklist
    
- [X] My code follows the style guidelines for this project
- [X] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [X] I have performed a self review of my own code
- [X] I have commented my code PARTICULARLY in hard to understand areas
- [X] I have added thorough tests where needed

### CODE REVIEW

File: src/apps/vpn/models/serverdata.cpp [hunk 1]
```diff
@@ -208,6 +209,51 @@ void ServerData::changeServer(const QString& countryCode,
...
+          mozilla::glean::sample::UserChangedEndpointGeoExtra{
+              ._server = "MultiHopEntry",
+          });
+    }
+  }
+
+  update(countryCode, cityName, entryCountryCode, entryCityName);
+}
+
+void ServerData::changeServerForTutorial(const QString& countryCode,
```
Comment (brizental) [inline src/apps/vpn/models/serverdata.cpp:246]: This feels like an unnecessary change given tutorials as a feature are about to go away.

File: src/apps/vpn/ui/screens/getHelp/developerMenu/ViewDeveloperMenu.qml [hunk 2]
```diff
@@ -290,6 +290,15 @@ MZViewBase {
...
         }
 
+        MZButton {
+            id: resetTipsAndTricksModal
+
+            text: "Reset Intro Tips and Tricks Modal"
+            onClicked: {
+                MZSettings.tipsAndTricksIntroShown = false
+            }
+        }
```
Comment (brizental) [inline src/apps/vpn/ui/screens/getHelp/developerMenu/ViewDeveloperMenu.qml:300]: This also feels unnecessary given the whole tips and tricks page is planned to be removed soon.

**Review by brizental — COMMENTED** [review]
Was this a full QA verification of metrics? I don't recall, for example, `tipsAndTricksViewOpened` being part of super dooper.

**Comment by mcleinman** [comment]
> Was this a full QA verification of metrics? I don't recall, for example, `tipsAndTricksViewOpened` being part of super dooper.

This was intended to be a full QA of all metrics getting sent in intended pings. 

Per the [PRD](https://mozilla-hub.atlassian.net/wiki/spaces/SECPRV/pages/117309748/Metrics+Collection+Dashboards+PRD), under education: `# ‘VPN tips and tricks’ view opened`.

The [QA sheet](https://docs.google.com/spreadsheets/d/1EdPD4bbSWW4VVfWPcQt4YqSB_Zpecfu1_f_5DKALAdw/edit#gid=281079299) of the doc comes from the `Detailed activity breakdown` sheet, which in turn comes from [the PRD](https://mozilla-hub.atlassian.net/wiki/spaces/SECPRV/pages/117309748/Metrics+Collection+Dashboards+PRD).

File: src/apps/vpn/models/serverdata.cpp (hunk 1 above)
Comment (mcleinman) [reply src/apps/vpn/models/serverdata.cpp:246]: Without this, we're going to count tutorial runs as server changes, which is not what we want (I think). I'd rather include this code than risk counting this wrong in 2.17 (if Super Dooper is live in 2.17, but for some reason tutorials are not removed until 2.18).

File: src/apps/vpn/ui/screens/getHelp/developerMenu/ViewDeveloperMenu.qml (hunk 2 above)
Comment (mcleinman) [reply src/apps/vpn/ui/screens/getHelp/developerMenu/ViewDeveloperMenu.qml:300]: Do you think there is a harm to including this until we remove tips and tricks? Without this, the only way to test this is to uninstall the app completely, it seemed helpful for testing to have a way to do this without having to log back in. I recognize this code may be pulled in the next few months, but still seems helpful for now.

**Comment by brizental** [comment]
> This was intended to be a full QA of all metrics getting sent in intended pings.
> 
> Per the [PRD](https://mozilla-hub.atlassian.net/wiki/spaces/SECPRV/pages/117309748/Metrics+Collection+Dashboards+PRD), under education: # ‘VPN tips and tricks’ view opened.
> 
> The [QA sheet](https://docs.google.com/spreadsheets/d/1EdPD4bbSWW4VVfWPcQt4YqSB_Zpecfu1_f_5DKALAdw/edit#gid=281079299) of the doc comes from the Detailed activity breakdown sheet, which in turn comes from [the PRD](https://mozilla-hub.atlassian.net/wiki/spaces/SECPRV/pages/117309748/Metrics+Collection+Dashboards+PRD).

Makes sense. Sorry for the oversight.

**Review by brizental — APPROVED** [review]

**Comment by mcleinman** [comment]
> > This was intended to be a full QA of all metrics getting sent in intended pings.
> > Per the [PRD](https://mozilla-hub.atlassian.net/wiki/spaces/SECPRV/pages/117309748/Metrics+Collection+Dashboards+PRD), under education: # ‘VPN tips and tricks’ view opened.
> > The [QA sheet](https://docs.google.com/spreadsheets/d/1EdPD4bbSWW4VVfWPcQt4YqSB_Zpecfu1_f_5DKALAdw/edit#gid=281079299) of the doc comes from the Detailed activity breakdown sheet, which in turn comes from [the PRD](https://mozilla-hub.atlassian.net/wiki/spaces/SECPRV/pages/117309748/Metrics+Collection+Dashboards+PRD).
> 
> Makes sense. Sorry for the oversight.

No worries. I wasn't sure, it was good for me to go back and check - it's been so long since I've thought about some of these pieces.

**Comment by mcleinman** [comment]
Working on fixing this up, something broke w/ the changed directory structure and a file is missing from the tests. Will pick it up on Monday.

**Comment by mcleinman** [comment]
Closing in favor of https://github.com/mozilla-mobile/mozilla-vpn-client/pull/7961
